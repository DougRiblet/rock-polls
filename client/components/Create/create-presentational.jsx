// @flow

import React from 'react';

type Props = {
  // eslint-disable-next-line no-undef
  createNewPoll: (poll: newPoll) => mixed,
  user_id: string,
  username: string,
};

type State = {
  question: string,
  answers: Array<string>,
};

export default class Create extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      question: '',
      answers: [],
    };
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChangeUsername = this.handleChangeUsername.bind(this);
    this.handleChangePassword = this.handleChangePassword.bind(this);
  }

  handleSubmit: Function;
  handleChangeUsername: Function;
  handleChangePassword: Function;

  // eslint-disable-next-line no-undef
  handleChangeUsername(event: SyntheticInputEvent<*>) {
    this.setState({ username: event.target.value });
  }

  // eslint-disable-next-line no-undef
  handleChangePassword(event: SyntheticInputEvent<*>) {
    this.setState({ password: event.target.value });
  }

  // eslint-disable-next-line no-undef
  handleSubmit(event: SyntheticInputEvent<*>) {
    event.preventDefault();
    const pollObj = {
      user_id: this.props.user_id,
      username: this.props.username,
      question: this.state.question,
      answers: this.state.answers,
    }
    this.props.createNewPoll(pollObj);
  }

  render() {
    return (
      <div id='create'>
        <h3>Create New Poll</h3>
        <form onSubmit={this.handleSubmit}>
          <label>
            Question:
            <input
            />
          </label>
          <br />
          <label>
            Answers:
            <input
            />
          </label>
          <br />
          <input type='submit' value='submit' />
        </form>
        <div className='show-status'>
          <p>Username: {this.props.username}</p>
          <p>User ID: {this.props.user_id}</p>
        </div>
      </div>
    );
  }
}
