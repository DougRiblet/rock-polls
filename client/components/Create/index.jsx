import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { createNewPoll } from '../../actions/actions';
import createPresent from './create-presentational';

const mapStateToProps = state => ({
  user_id: state.user_id,
  username: state.username,
});

const mapDispatchToProps = dispatch => bindActionCreators({ createNewPoll }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(createPresent);

