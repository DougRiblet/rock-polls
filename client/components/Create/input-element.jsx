// @flow

import React from 'react';

const InputEl = ({id}) => {
  <label>
    <input
      id={id}
      type='text'
      ref={(input) => { this.textInput = input; }}
      value={this.state}
      maxLength='90'
      size='90'
      onChange={this.handleChangeUsername}
    />
  </label>
};

export default InputEl;
