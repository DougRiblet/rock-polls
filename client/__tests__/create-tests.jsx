import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Create from '../components/Create/create-presentational';

jest.mock('random-id', () => {
  let val = 0;
  return () => {
    val += 1;
    return `abcdefghijklmnop${val}`;
  };
});

describe('Create', () => {
  test('Snapshot', () => {
    const tree = renderer.create(<Create
      createNewPoll={() => jest.fn()}
      user_id=''
      username=''
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Create
      createNewPoll={() => jest.fn()}
      user_id=''
      username=''
    />, div);
  });
});
