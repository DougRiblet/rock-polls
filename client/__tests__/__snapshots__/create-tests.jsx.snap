// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Snapshot 1`] = `
<div
  id="create"
>
  <div>
    <h3>
      Create New Poll
    </h3>
    <h4
      className="create-label"
    >
      Question:
    </h4>
    <label>
      <textarea
        cols="90"
        id="question-input"
        maxLength="390"
        onChange={[Function]}
        rows="5"
        value=""
      />
    </label>
    <br />
    <h4
      className="create-label"
    >
      Answers:
    </h4>
    <label>
      <input
        className="answer-input"
        maxLength="99"
        name="5zdSnO7GqEpHqCTRUg"
        onChange={[Function]}
        size="80"
        value=""
      />
    </label>
    <label>
      <input
        className="answer-input"
        maxLength="99"
        name="4V1hPhGWVk92qQ4wmF"
        onChange={[Function]}
        size="80"
        value=""
      />
    </label>
    <div
      className="create-buttons"
    >
      <button
        onClick={[Function]}
      >
        Add Answer Field
      </button>
      <button
        onClick={[Function]}
      >
        Preview Poll
      </button>
    </div>
    <div
      className="show-status"
    >
      <p>
        Username: 
        
      </p>
      <p>
        User ID: 
        
      </p>
    </div>
  </div>
</div>
`;
